tracing-opentelemetry/src/main/java/io/micronaut/tracing/DefaultOpenTelemetryFactory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing;
+
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.context.annotation.Property;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import jakarta.inject.Singleton;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static io.micronaut.runtime.ApplicationConfiguration.APPLICATION_NAME;
+
+/**
+ * Registers an OpenTelemetry bean.
+ *
+ * @author Nemanja Mikic
+ */
+@Factory
+public class DefaultOpenTelemetryFactory {
+
+    private static final String SERVICE_NAME_KEY = "otel.service.name";
```

Comment:
Can we have `opentelemetry.` instead of `otel.`? WDYT @graemerocher 
I would also have a constant `PREFIX`.

Reply:
The "otel" prefix comes from https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md and this factory overrides some defaults of this library.

Reply:
Agree, I think if we want to reuse the original official OpenTelemetry configuration properties we should use `otel` as well. Maybe it will be nice to support the configuration of most of [these properties](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md) directly in the `application.yml` file.

Reply:
@donbeave We already support most of linked properties through `application.yml`

Reply:
Fine with me